infra/common/src/test/java/org/apache/shardingsphere/infra/metadata/database/schema/reviser/table/TableMetadataReviseEngineTest.java

```diff
@@ -0,0 +1,135 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.metadata.database.schema.reviser.table;
+
+import org.apache.shardingsphere.infra.database.core.metadata.data.model.ColumnMetaData;
+import org.apache.shardingsphere.infra.database.core.metadata.data.model.IndexMetaData;
+import org.apache.shardingsphere.infra.database.core.metadata.data.model.TableMetaData;
+import org.apache.shardingsphere.infra.database.core.type.DatabaseType;
+import org.apache.shardingsphere.infra.metadata.database.schema.reviser.MetaDataReviseEntry;
+import org.apache.shardingsphere.infra.rule.ShardingSphereRule;
+
+import org.junit.jupiter.api.BeforeEach;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import javax.sql.DataSource;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TableMetadataReviseEngineTest<T extends ShardingSphereRule> {
+    
+    @Mock
+    private T mockRule;
+    
+    @Mock
+    private DatabaseType mockDatabaseType;
+    
+    @Mock
+    private DataSource mockDataSource;
+    
+    @Mock
+    private MetaDataReviseEntry<T> mockMetaDataReviseEntry;
+    
+    @InjectMocks
+    private TableMetaDataReviseEngine<T> mockTableMetaDataReviseEngine;
+    
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+    
+    /**
+     * Testing revise method with TableNameReviser.
+     */
+    @Test
+    public void testReviseWithTableNameReviser() {
+        // Arrange
+        TableNameReviser<T> tableNameReviser = mock(TableNameReviser.class);
+        
+        // Create a sample original TableMetaData
+        TableMetaData originalMetaData = new TableMetaData(
+                "originalTableName",
+                new ArrayList<>(),
+                null,
+                null);
+        
+        // Act
+        doReturn(Optional.of(tableNameReviser)).when(mockMetaDataReviseEntry).getTableNameReviser();
+        when(tableNameReviser.revise(anyString(), eq(mockRule))).thenReturn("revisedTableName");
+        
+        // Call the revise method
+        TableMetaData revisedMetaData = mockTableMetaDataReviseEngine.revise(originalMetaData);
+        
+        // Assert
+        // Verify that the revised table name is returned
+        assertEquals("revisedTableName", revisedMetaData.getName());
```

Comment:
Please use assertThat replace to assertEquals

Reply:
Ack